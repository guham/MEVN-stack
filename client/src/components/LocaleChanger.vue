<template>
  <div class="inline-block relative w-24 mr-2 ">
    <select
      id="locale-changer"
      v-model="$i18n.locale"
      class="select-standard w-full"
    >
      <option
        v-for="(lang, i) in locales"
        :key="`Lang${i}`"
        :value="lang"
      >
        {{ lang.toUpperCase() }}
      </option>
    </select>
    <div class="pointer-events-none absolute pin-y pin-r flex items-center px-2 text-grey-darker">
      <svg
        class="fill-current h-4 w-4"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 20 20"
      >
        <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
      </svg>
    </div>
  </div>
</template>

<script>
import { locales } from '@/i18n';
import { setLocaleCookie } from '@/services/cookies';

export default {
  name: 'LocaleChanger',
  data: () => ({
    locales,
  }),

  watch: {
    '$i18n.locale': function change(locale) {
      setLocaleCookie(locale);
    },
  },
};
</script>
