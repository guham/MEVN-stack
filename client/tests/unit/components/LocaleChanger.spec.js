import LocaleChanger from '@/components/LocaleChanger.vue';
import factory from '../factory';

describe('LocaleChanger.vue', () => {
  test('should change locale as expected', () => {
    const wrapper = factory(LocaleChanger);
    expect(wrapper.vm.$i18n.locale).toBe('en');
    const options = wrapper.find('select').findAll('option');
    options.at(1).setSelected();
    expect(wrapper.vm.$i18n.locale).toBe('fr');
  });

  test('has the expected html structure', () => {
    const wrapper = factory(LocaleChanger);
    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });
});
