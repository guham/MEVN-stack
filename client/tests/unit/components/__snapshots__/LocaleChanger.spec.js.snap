// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleChanger.vue has the expected html structure 1`] = `
<div class="inline-block relative w-24 mr-2 "><select id="locale-changer" class="select-standard w-full">
    <option value="en">EN</option>
    <option value="fr">FR</option>
    <option value="es">ES</option>
  </select>
  <div class="pointer-events-none absolute pin-y pin-r flex items-center px-2 text-grey-darker"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" class="fill-current h-4 w-4">
      <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"></path>
    </svg></div>
</div>
`;
